File: c/caesar.c
#include <stdio.h>
#include <stdlib.h>
#include <ctype.h>

int main(int argc, char *argv[]) {
    if (argc < 3) {
        fprintf(stderr, "usage: %s shift text\n", argv[0]);
        return 1;
    }
    int shift = atoi(argv[1]);
    for (int i = 2; i < argc; i++) {
        char *p = argv[i];
        if (i > 2) putchar(' ');
        for (; *p; p++) {
            if (isalpha((unsigned char)*p)) {
                char base = isupper((unsigned char)*p) ? 'A' : 'a';
                putchar(((*p - base + shift) % 26) + base);
            } else {
                putchar(*p);
            }
        }
    }
    putchar('\n');
    return 0;
}

File: py/caesar.py
import sys

def caesar(text, shift):
    res = []
    for ch in text:
        if ch.isalpha():
            base = ord('A') if ch.isupper() else ord('a')
            res.append(chr((ord(ch) - base + shift) % 26 + base))
        else:
            res.append(ch)
    return ''.join(res)

if __name__ == '__main__':
    if len(sys.argv) < 3:
        print('usage: python caesar.py shift text')
        sys.exit(1)
    shift = int(sys.argv[1])
    text = ' '.join(sys.argv[2:])
    print(caesar(text, shift))
